tests/test_stomp.py

```diff
@@ -63,3 +66,59 @@ def test_stomp_A_B_join(T_A, T_B):
     utils.replace_inf(left)
     utils.replace_inf(right)
     npt.assert_almost_equal(left, right)
+
+
+@pytest.mark.parametrize("T_A, T_B", test_data)
+@pytest.mark.parametrize("substitute_B", substitution_values)
+def test_stomp_nan_inf_self_join(T_A, T_B, substitute_B):
+    m = 3
+
+    T_B_sub = T_B.copy()
+
+    substitution_locations = [slice(0, 0), 0, -1, slice(1, 3), [0, 3]]
+    for substitution_location_B in substitution_locations:
+        T_B_sub[:] = T_B[:]
+        T_B_sub[substitution_location_B] = substitute_B
+
+        zone = int(np.ceil(m / 4))
+        left = np.array(
+            [
+                utils.naive_mass(Q, T_B_sub, m, i, zone, True)
+                for i, Q in enumerate(core.rolling_window(T_B_sub, m))
+            ],
+            dtype=object,
+        )
+        right = stomp(T_B_sub, m, ignore_trivial=True)
+        utils.replace_inf(left)
+        utils.replace_inf(right)
+        npt.assert_almost_equal(left, right)
```

Comment:
This is great. This implies that our tests are deterministic and that the matrix profile indices are correct!